core/src/main/java/org/apache/brooklyn/core/typereg/BrooklynCatalogBundleResolvers.java

```diff
@@ -0,0 +1,196 @@
+            if (LOG.isTraceEnabled()) {
+                LOG.trace("SCORE for '" + input + "' at " + transformer + ": " + score);
+            }
+            if (score>0) byScoreMulti.put(score, transformer);
+        }
+        Map<Double, Collection<BrooklynCatalogBundleResolver>> tree = new TreeMap<Double, Collection<BrooklynCatalogBundleResolver>>(byScoreMulti.asMap());
+        List<Collection<BrooklynCatalogBundleResolver>> highestFirst = new ArrayList<Collection<BrooklynCatalogBundleResolver>>(tree.values());
+        Collections.reverse(highestFirst);
+        return ImmutableList.copyOf(Iterables.concat(highestFirst));
+    }
+
+    public static ReferenceWithError<OsgiBundleInstallationResult> install(ManagementContext mgmt, Supplier<InputStream> input,
+                                                                           BundleInstallationOptions options) {
+        LOG.debug("Installing bundle {} / {}", input, (options==null ? null : options.knownBundleMetadata));
+        File fileToDelete = null;
+        if (input==null && options.knownBundleMetadata==null) {
+            return ReferenceWithError.newInstanceThrowingError(null, new IllegalArgumentException("Bundle contents or reference must be supplied"));
+        }
+        if (input==null && options.knownBundleMetadata!=null) {
+            // installing to brooklyn a bundle already installed to OSGi
+            PrepareInstallResult prepareResult = BrooklynBomOsgiArchiveInstaller.prepareInstall(mgmt, options.knownBundleMetadata, null, input, options.forceUpdateOfNonSnapshots, null);
+            if (prepareResult.resultObject != null) {
+                return ReferenceWithError.newInstanceWithoutError(prepareResult.resultObject);
+            }
+
+            fileToDelete = prepareResult.zipFile;
+            if (prepareResult.zipFile != null) {
+                input = InputStreamSource.of(prepareResult.zipFile.getName(), prepareResult.zipFile);
+            } else {
+                return ReferenceWithError.newInstanceThrowingError(null, new IllegalArgumentException("Bundle contents or known reference must be supplied; " +
+                        options.knownBundleMetadata + " not known"));
+            }
+        }
+
+        OsgiBundleInstallationResult firstResult = null;
+        try {
+            List<BrooklynCatalogBundleResolver> resolvers = forBundle(mgmt, input, options);
+            Collection<String> resolversWhoDontSupport = new ArrayList<String>();
+            Collection<Exception> failuresFromResolvers = new ArrayList<Exception>();
+            for (BrooklynCatalogBundleResolver t : resolvers) {
+                try {
+                    ReferenceWithError<OsgiBundleInstallationResult> result = t.install(input, options);
+                    if (result == null) {
+                        resolversWhoDontSupport.add(t.getFormatCode() + " (returned null)");
```

Comment:
you don't log this anywhere below, didn't you want to include this in a message somewhere?

Reply:
again i just copied but i think yes, in both cases we want to include that info in a log message